new-website/src/components/location/location-table.jsx

```diff
@@ -2,51 +2,93 @@ import React, { useState } from "react"
-            onError({ code: "no-permission" })
-            throw new Error("Missing Partner Id")
-        }
-        const { title, partnerId } = data
-        if (!title || title.length < 0) {
-            onError({ code: "requiredFieldMissing/title" })
-            throw new Error("missing title")
+function addNewRow(userPartnerIds, allPartnerConfigs, onError) {
+    const newPartnerLocationDoc = async () => {
+        // TODO: The partner name is a prefix in the location ID.
+        // Not sure this is ideal, but it is what we have currently.
+        const partnerId = choosePartnerId(null, userPartnerIds, onError)
+        const partnerConfig = allPartnerConfigs.find(p => p.id == partnerId)
+
+        // ShortID may collide with existing entry, retry a few times
+        for (let i=0; i<10; i++) {
+            const docId = `${partnerConfig.name}-${shortid.generate()}`
+            const docRef = firebase.firestore().collection(LOCATION_COLLECTION).doc(docId)
+            const doc = await docRef.get().catch(onError)
+            if (!doc.exists) {
+                docRef.set({ partnerId: partnerId })
+                return docId;
+            }
         }
+        onError({ code: "newPartnerLocationDoc/failed" })
+        throw new Error("Create new doc failed")
+    }
+
+    return async (event) => {
+        // For now, let's use a single page for add and edit
+        const docId = await newPartnerLocationDoc()
+        navigate(`/partner/location/${docId}`)
 
-        return firebase
-            .firestore()
-            .collection(LOCATION_COLLECTION)
-            .add({ ...data, partnerId: choosePartnerId(partnerId, userPartnerIds) })
-            .catch(onError)
+        // Upon confirm in add page:
+        //if (!userPartnerIds || userPartnerIds.length <= 0) {
+        //    onError({ code: "no-permission" })
+        //    throw new Error("Missing Partner Id")
+        //}
+        //const { title, partnerId } = data
+        //if (!title || title.length < 0) {
+        //    onError({ code: "requiredFieldMissing/title" })
+        //    throw new Error("missing title")
+        //}
+        //
+        //return firebase
+        //    .firestore()
+        //    .collection(LOCATION_COLLECTION)
+        //    .add({ ...data, partnerId: choosePartnerId(partnerId, userPartnerIds) })
+        //    .catch(onError)
     }
 }
 
 function deleteRow(onError) {
-    return async function deleteIt({ id }) {
+    return async ({ id }) => {
         return firebase.firestore().collection(LOCATION_COLLECTION).doc(id).delete().catch(onError)
     }
 }
 
 function updateRow(onError) {
-    return async function deleteIt({ id, ...rest }) {
-        return firebase
-            .firestore()
-            .collection(LOCATION_COLLECTION)
-            .doc(id)
-            .update({ ...rest })
-            .catch(onError)
+    return async (event, rowData) => {
+        navigate(`/partner/location/${rowData.id}`)
```

Comment:
How would I implement the function that handles the navigation target here? Not via `firebase.json` right?

Reply:
Nope :D ... you can create a simple static page `src/pages/partner/location/add.jsx` and hand over the id as `location.state` property (https://www.gatsbyjs.org/docs/gatsby-link/#how-to-use-the-navigate-helper-function)

I would not try to generate for each location a separate page. no need

Reply:
Ok, handing it over as state sounds great. Is there a way we can still have a permanent address for each location?

Reply:
Seems like this is what I am looking for: https://www.gatsbyjs.org/docs/client-only-routes-and-user-authentication/

Reply:
..but I don't get it to work in our case 🙁